nx/source/services/news.c

```diff
@@ -0,0 +1,308 @@
+#define NX_SERVICE_ASSUME_NON_DOMAIN
+#include "services/news.h"
+#include "service_guard.h"
+#include "runtime/hosversion.h"
+#include <string.h>
+
+static NewsServiceType g_newsServiceType;
+static Service g_newsCreatorSrv;
+static Service g_newsSrv;
+
+static Result _newsCreateNewsService(Service *srv_out);
+
+NX_GENERATE_SERVICE_GUARD_PARAMS(news, (NewsServiceType service_type), (service_type));
+
+Result _newsInitialize(NewsServiceType service_type) {
+    Result rc;
+
+    if (service_type >= NewsServiceType_Count)
+        return MAKERESULT(Module_Libnx, LibnxError_BadInput);
+
+    const char *service_names[] = {
```

Comment:
```suggestion
    static const char* const service_names[] = {
```

Otherwise this is a mutable array that goes on the stack.